build_tools/buildkite/github/post_benchmarks_as_pr_comment.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+# Lint as: python3
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Posts benchmark results to GitHub as pull request comments.
+
+This script is meant to be used by Buildkite for automation. It requires the
+following environment to be set:
+
+- BUILDKITE_BUILD_URL: the link to the current buildkite build.
+- BUILDKITE_COMMIT: the pull request HEAD commit.
+- BUILDKITE_PULL_REQUEST: the current pull request number.
+- GITHUB_TOKEN: personal access token to authenticate against GitHub API.
+
+This script uses pip package "markdown_strings".
+
+Example usage:
+  # Export necessary environment variables:
+  export ...
+  # Then run the script:
+  python3 post_benchmarks_as_pr_comment.py <benchmark-json-file>..
+  #   where each <benchmark-json-file> is expected to be of format:
+  #   {"commit": <commit-sha>, "benchmarks": {<name>: <value>, ...}}
+"""
+
+import argparse
+import json
+import os
+import requests
+import markdown_strings as md
+
+GITHUB_IREE_API_PREFIX = "https://api.github.com/repos/google/iree"
+
+
+def get_required_env_var(var):
+  """Gets the value for a required environment variable."""
+  value = os.getenv(var, None)
+  if value is None:
+    raise RuntimeError(f'Missing environment variable "{var}"')
+  return value
+
+
+def aggregate_all_benchmarks(benchmark_files):
+  """Aggregates all benchmarks in the given files.
+
+  Args:
+  - benchmark_files: A list of JSON files, each with the format:
+    {"commit": <commit-sha>, "benchmarks": {<name>: <value>, ...}}
+
+  Returns:
+  - A list of benchmark (name, value) tuples.
+  """
+  pr_commit = get_required_env_var("BUILDKITE_COMMIT")
+  all_benchmarks = []
+
+  for benchmark_file in benchmark_files:
+    with open(benchmark_file) as f:
+      benchmarks = json.load(f)
+      if benchmarks["commit"] != pr_commit:
+        raise ValueError("Inconsistent pull request commit")
+      for name, value in benchmarks["benchmarks"].items():
+        if name in all_benchmarks:
+          raise ValueError(f"Duplicated benchmarks: {name}")
+        all_benchmarks.append((name, value))
```

Comment:
Nit: since you're done with the file at this point, I think you can unindent this block and close the file

```suggestion
    if benchmarks["commit"] != pr_commit:
      raise ValueError("Inconsistent pull request commit")
    for name, value in benchmarks["benchmarks"].items():
      if name in all_benchmarks:
        raise ValueError(f"Duplicated benchmarks: {name}")
      all_benchmarks.append((name, value))
```